server/src/main/java/org/elasticsearch/action/bulk/TransportSimulateBulkAction.java

```diff
@@ -360,6 +372,23 @@ private void validateUpdatedMappings(
                 0
             );
         });
+        final Collection<String> ignoredFields;
+        if (result == null) {
+            ignoredFields = List.of();
```

Comment:
We could probably just do `return List.of();` here and avoid having to create the `ignoredFields` local field? It's not a big deal either way though.

Reply:
I kind of prefer having a single return statement rather than 3 -- it makes debugging easier for me.

Reply:
I'm okay either way :)